File: app/src/main/java/com/fabienraye/nasawatcher/features/commons/presenters/BasePresenter.java
package com.fabienraye.nasawatcher.features.commons.presenters;


import rx.subscriptions.CompositeSubscription;

public abstract class BasePresenter<T extends IView> {
    protected T mView;

    // To manage observables in the presenter lifecycle
    protected CompositeSubscription mCompositeSubscription;

    public BasePresenter() {
        mCompositeSubscription = new CompositeSubscription();
    }

    public void onViewCreated() {}

    public void onViewStarted() {}

    public void onViewResumed() {}

    /**
     * View will be destroyed
     */
    public void onViewDestroyed() {
        mCompositeSubscription.unsubscribe();
    }
}

File: app/src/main/java/com/fabienraye/nasawatcher/features/commons/presenters/BasePresenter.kt
package com.fabienraye.nasawatcher.features.commons.presenters


import rx.subscriptions.CompositeSubscription

abstract class BasePresenter<T : IView> {

    protected var mView: T? = null

    // To manage observables in the presenter lifecycle
    protected var mCompositeSubscription: CompositeSubscription = CompositeSubscription()

    fun onViewCreated() {}

    fun onViewStarted() {}

    fun onViewResumed() {}

    /**
     * View will be destroyed
     */
    fun onViewDestroyed() {
        mCompositeSubscription.unsubscribe()
    }
}
